fdbserver/storageserver.actor.cpp

```diff
@@ -2683,25 +2644,47 @@ MutationsAndVersionRef filterMutations(Arena& arena,
 			if (!modifiedMutations.present() &&
 			    (m.mutations[i].param1.compareSuffix(range.begin, commonPrefixLength) < 0 ||
 			     m.mutations[i].param2.compareSuffix(range.end, commonPrefixLength) > 0)) {
-				modifiedMutations = m.mutations.slice(0, i);
+				if (encrypted && m.encrypted.present()) {
+					modifiedMutations = m.encrypted.get().slice(0, i);
+				} else {
+					modifiedMutations = m.mutations.slice(0, i);
+				}
 				arena.dependsOn(range.arena());
 			}
 			if (modifiedMutations.present()) {
 				// param1 < range.end && range.begin < param2
 				if (m.mutations[i].param1.compareSuffix(range.end, commonPrefixLength) < 0 &&
 				    range.begin.compareSuffix(m.mutations[i].param2, commonPrefixLength) < 0) {
-					modifiedMutations.get().push_back(arena,
-					                                  MutationRef(MutationRef::ClearRange,
-					                                              std::max(range.begin, m.mutations[i].param1),
-					                                              std::min(range.end, m.mutations[i].param2)));
+					StringRef clearBegin = m.mutations[i].param1;
+					StringRef clearEnd = m.mutations[i].param2;
+					bool modified = false;
+					if (clearBegin.compareSuffix(range.begin, commonPrefixLength) < 0) {
+						clearBegin = range.begin;
+						modified = true;
+					}
+					if (range.end.compareSuffix(clearEnd, commonPrefixLength) < 0) {
+						clearEnd = range.end;
+						modified = true;
+					}
+					if (modified) {
+						// FIXME: encrypt
```

Comment:
Food for thought: Given per-tenant encryption domain semantics, we may have to split the ClearRange across tenant-boundaries before encrypting the modifications.

Maybe useful to record it as FIXME statement?

Reply:
Each change feed should currently only belong to a single tenant, so I don't think that applies here. We don't enforce this strictly at this level though

Reply:
That was my thought process as well, however, wasn't aware how the enforcement is done. If we don't do it, how is this invariant guaranteed?